### ISSUE TITLE
Flax StableDiffusion models give wrong from_config deprecation warning

### ISSUE THREAD

**Issue by krahnikblis** [huggingface/diffusers#4723]
### Describe the bug

loading models using `from_pretrained` causes a warning that `from_config` is deprecated. but i'm not calling `from_config`. happens with VAE, Unet, and Scheduler.

suggestion on the warning - should indicate what to replace the supposed `from_config` call with, to be future compatible. i.e., different method? dictionary of config in same method? 

### Reproduction

```python
params = {}
dtype = jnp.bfloat16
# vae
use_mse_vae_weights = True
vae_folder = None
vae_path = "/content/drive/My Drive/Diffusion/sd-vae-ft-mse/"
from_pt = (use_mse_vae_weights and list(pathlib.Path(vae_path + (vae_folder or "")).glob("*.msgpack")) == [])
vae, params["vae"] = jax.device_get(FlaxAutoencoderKL.from_pretrained(vae_path, subfolder=vae_folder, dtype=dtype, from_pt=from_pt))
```

this produces this warning:
```/usr/local/lib/python3.10/dist-packages/diffusers/configuration_utils.py:239: FutureWarning: It is deprecated to pass a pretrained model name or path to `from_config`.
  deprecate("config-passed-as-path", "1.0.0", deprecation_message, standard_warn=False)```

### System Info

colab pro, TPU high-RAM

### Who can help?

@patrickvonplaten @sayakpaul @williamberman @pcuenca

**Comment by patrickvonplaten** [comment]
cc @pcuenca could you check here?
